### PR TITLE
fix: adds fallback for Safari < 14 and older browsers

### PR DESCRIPTION
**Pull request by anteriovieira**
Hi @antfu , I came across problems with the media query during some tests and decided to send my proposal.

### CODE REVIEW

File: packages/core/useMediaQuery/index.ts [hunk 1]
```diff
@@ -15,11 +15,20 @@ export function useMediaQuery(query: string) {
...
+      mediaQuery.addListener(handler)
+    }
+  }
+
+  registerEvent()
 
-  tryOnUnmounted(() => {
-    mediaQuery.addEventListener('change', handler)
-  })
+  tryOnUnmounted(registerEvent)
```
Comment (antfu) [inline packages/core/useMediaQuery/index.ts:31]: Oh, this should be `removeEventListener`

Comment (anteriovieira) [reply packages/core/useMediaQuery/index.ts:31]: Sorry, can you please give me an example?

Comment (anteriovieira) [reply packages/core/useMediaQuery/index.ts:31]: Oh, I see hehe

File: packages/core/useMediaQuery/index.ts [hunk 2]
```diff
@@ -15,11 +15,20 @@ export function useMediaQuery(query: string) {
...
+      mediaQuery.addEventListener('change', handler)
+    }
+    catch (e) {
+      // @ts-ignore
+      // Fallback for Safari < 14 and older browsers
+      mediaQuery.addListener(handler)
+    }
+  }
+
+  registerEvent()
```
Comment (antfu) [inline packages/core/useMediaQuery/index.ts:29]:
```suggestion
  try {
    mediaQuery.addEventListener('change', handler)
  }
  catch (e) {
    // @ts-ignore
    // Fallback for Safari < 14 and older browsers
    mediaQuery.addListener(handler)
  }
```

File: packages/core/useMediaQuery/index.ts (hunk 1 above)
Comment (antfu) [inline packages/core/useMediaQuery/index.ts:31]:
```suggestion
  tryOnUnmounted(() => {
  try {
    mediaQuery.removeEventListener('change', handler)
  }
  catch (e) {
    // @ts-ignore
    // Fallback for Safari < 14 and older browsers
    mediaQuery.removeListener(handler)
  }
  })
```

**Review by antfu — CHANGES_REQUESTED** [review]

**Comment by antfu** [comment]
I am not sure about using try...catch for this. Maybe 

```ts
if ('addEventListener' in mediaQuery) {
  mediaQuery.addEventListener('change', handler)
}
else {
  mediaQuery.addListener('change', handler)
}
```

File: packages/core/useMediaQuery/index.ts [hunk 3]
```diff
@@ -15,10 +15,22 @@ export function useMediaQuery(query: string) {
     matches.value = event.matches
   }
 
-  mediaQuery.addEventListener('change', handler)
+  if ('addEventListener' in mediaQuery) {
+    mediaQuery.addEventListener('change', handler)
+  }
+  else {
+    // @ts-ignore
```
Comment (antfu) [inline packages/core/useMediaQuery/index.ts:22]: Let's keep your original comment 😉

Comment (anteriovieira) [reply packages/core/useMediaQuery/index.ts:22]: done! 😃

**Comment by antfu** [comment]
I have invited you to be the collaborator of this project. Thanks for your work!

**Comment by anteriovieira** [comment]
Thank you very much, I am really excited about all your work. I hope to be able to contribute to the growth of this project. 🚀

**Pull request merged**